### ISSUE TITLE
Is it possible to do error handling when the user inputs a wrong number of parameters?

### ISSUE THREAD

**Issue by mahnamizjaf** [ThePhD/sol2#910]
lets say I've got a function "foo"  in C++ that I want to call in lua

my script does this :

```
function update()
--this is supposedly wrong
    foo()
end
```
and this is the C++ code:
```
struct myClass 
{
  int haha;
    myClass()
      :haha{0}{};
}
void foo(myClass mClass)
{
  std::cout<<mClass.haha<<std::endl;
}
int main()
{
    sol::state mLua;
    mLua.open_libraries(sol::lib::base, sol::lib::package, 
					 sol::lib::string, sol::lib::os, 
					 sol::lib::math, sol::lib::table, 
					 sol::lib::io);

    mLua.new_usertype<myClass>("myClass",sol::constructors< myClass()>());

    mLua["foo"] = &foo;
    sol::environment& env =sol::environment{ mLua, sol::create, mLua.globals() };
    mLua.script_file("someLua.lua",env);
    mLua["foo"] = &foo;
    env["update"]();
}
```

technically this is supposed to be wrong as "foo" takes in a "myClass" and when i tested this out I got a segmentation fault.
however upon further experimenting if i change my "foo" function to take in primitive type
(e.g.`foo(int haha)`).
the function runs properly without crashing.

TLDR; is there a way for me to show debugging errors or handle them in runtime if the user were to input the wrong number/type of parameters?

**Comment by ThePhD** [comment]
https://sol2.readthedocs.io/en/latest/safety.html#config

https://github.com/ThePhD/sol2/blob/develop/examples/source/tutorials/quick_n_dirty/running_lua_code_safely.cpp#L36

https://sol2.readthedocs.io/en/latest/api/protected_function.html

**Comment by frankhale** [comment]
I am in this same boat as well. If my C++ function is called from Lua and Lua does not pass in enough parameters the function isn't actually called on the C++ side but there are no errors that I can report back to the script writer. I've looked at all those links above. I cannot figure out how to get Sol3 to actually report an error to the console. I've Googled but there is basically nothing out there except the docs mentioned above. I'll open a new issue.
